src/main/java/com/microsoft/sqlserver/jdbc/SQLServerClob.java

```diff
@@ -328,7 +328,13 @@ public String getSubString(long pos, int length) throws SQLException {
     public long length() throws SQLException {
         checkClosed();
         if (null == value && activeStreams.get(0) instanceof BaseInputStream) {
-            return (long) ((BaseInputStream) activeStreams.get(0)).payloadLength;
+            int length = ((BaseInputStream) activeStreams.get(0)).payloadLength;
+            if (null != typeInfo) {
+                String columnTypeName = typeInfo.getSSTypeName();
+                return ("nvarchar".equalsIgnoreCase(columnTypeName)
+                        || "ntext".equalsIgnoreCase(columnTypeName)) ? length / 2 : length;
```

Comment:
SSType's name can return nchar as well as well as nvarchar and ntext. Should we also cover for nchar here?

Reply:
We don't claim to support Char/NChar for clob/nclob: https://docs.microsoft.com/en-us/sql/connect/jdbc/using-advanced-data-types?view=sql-server-ver15